nearcore/src/config.rs

```diff
@@ -1161,12 +1162,33 @@ async fn download_file_impl(
     let https_connector = hyper_tls::HttpsConnector::new();
     let client = hyper::Client::builder().build::<_, hyper::Body>(https_connector);
     let mut resp = client.get(uri).await.map_err(FileDownloadError::HttpError)?;
+    let bar = match resp.size_hint().upper() {
+        Some(file_size) => {
+            let bar = ProgressBar::new(file_size);
+            bar.set_style(
+                ProgressStyle::default_bar().template(
+                    "{spinner:.green} [{elapsed_precise}] [{wide_bar:.cyan/blue}] {bytes}/{total_bytes} [{bytes_per_sec}] ({eta})"
+                ).progress_chars("#>-")
+            );
+            bar
+        }
+        None => {
+            let bar = ProgressBar::new_spinner();
+            bar.set_style(
+                ProgressStyle::default_bar()
+                    .template("{spinner:.green} [{elapsed_precise}] {bytes} [{bytes_per_sec}]"),
+            );
+            bar
+        }
+    };
     while let Some(next_chunk_result) = resp.data().await {
         let next_chunk = next_chunk_result.map_err(FileDownloadError::HttpError)?;
         file.write_all(next_chunk.as_ref())
             .await
             .map_err(|e| FileDownloadError::WriteError(path.to_path_buf(), e))?;
+        bar.inc(next_chunk.len() as u64);
     }
+    bar.finish();
```

Comment:
I’m assuming it’s ok not ta call `bar.finish()` in case of an early return?

Reply:
It will just leave an unfinished progress bar like this one which I think is fine:

![latest-screenshot](https://user-images.githubusercontent.com/7879134/148890754-aa5e5ad7-cb80-4676-8c69-21b4cdd2baa7.png)